lnrpc/rpc.proto

```diff
@@ -461,6 +461,14 @@ message SendRequest {
     payment to the recipient.
     */
     string payment_request = 6;
+
+    // Cut off specifying a fee value above which path finding will be short circuited
+    oneof fee_cut_off {
```

Comment:
I could imagine a user wanting to specify both.

Reply:
Interesting, can I have a for instance?

Reply:
I just don't see the options as mutually exclusive. A user might want to say "I will pay up to 20% of the transaction value in fees with a hard limit of 0.1 BTC."

It wouldn't complicate the implementation much either. You just set max `fee = min(absolute cutoff, ratio cutoff * amount)`.

Reply:
I would argue that this logic could be moved to the caller, such that we can have only the `limit_satoshis` field, and make the caller do the necessary calculation. That would simplify the code a bit?

Reply:
Yeah, that could be the way to go. I don't think the code is overly complex as it is, but it would simplify it a bit. Would certainly simplify RPC testing though.

Reply:
Should this limit be specified in millisatoshis? The issue says satoshis, but I think that removes a ton of granularity between 0 and 1 satoshi. Default fee is 1 millisatoshi, correct?

Reply:
@halseth I am attempting to make this a singular field as you described, but it seems that proto3 provides no way to distinguish between unset fields and fields that were explicitly set to 0 (i.e. unset fields are set to their 0 value). With the oneof field, it was simple to first check against `nil`. Any guidance on how to overcome this? Should we require the caller to always set this value if they would like routes with fee > 0? 

On second thought, I guess that would be a useful function, so we would look for fee-less routes by default, and if none exist then the user could set their real max fee. It seems more intuitive for unset to mean "don't care", rather than 0, though.

Let me know what you think!

Reply:
I was thinking maybe require the caller to set its `maxfee`. The only downside I see by doing this, is that it might break apps that are already using this API (since maxFee would default to 0 and they would struggle to find 0 fee routes). But I would argue that's a change you can do in an alpha release 😄 

Also remember that you must update `lncli` (found in `cmd/`) to accept the new parameters, and use the new gRPC fields.